src/tsx/coreConnection.tsx

```diff
@@ -1,31 +1,49 @@
+import { INodeCollection } from "./types";
+
 /**
  * This class handles the communication with the core.
  * The user calls methods directly to
  * send a message to the core, and can register for events.
  */
 interface ICoreConnection extends EventTarget {
   // UI-initiated actions
-  /** Example of UI-initiated action */
-  askForIncrement(): void;
+  handshake(): void;
+  nextNode(): void;
 
   // Events
-  addEventListener(event: "increment", listener: () => void): void;
+  addEventListener(event: "nodes",
+                   listener: (event: CustomEvent<INodeCollection>) => void): void;
+  addEventListener(event: "current-node",
+                   listener: (event: CustomEvent<string>) => void): void;
 }
 
 /**
  * This is a dummy implementation of a connection
  * to the core, for use in development.
  */
 class DummyCoreConnection extends EventTarget implements ICoreConnection {
+  private nodes: INodeCollection;
+  private currentNode: string;
 
   // We only use the address in the real core connection
-  // tslint:disable-next-line no-empty
   constructor(address: string) {
     super();
+    this.nodes = {
+      a: {next: "b", prompt: "Välkommen till Kulturkrocks senaste föreställning!"},
+      b: {next: "c", prompt: "Jag ser att ni fortfarande är på väg in."},
+      c: {next: "a", prompt: "Några av er kanske missade det, så jag säger det igen:"},
+    };
+    this.currentNode = "a";
+  }
+
+  public handshake() {
+    this.dispatchEvent(new CustomEvent("nodes", {detail: this.nodes}));
+    this.dispatchEvent(new CustomEvent("current-node", {detail: this.currentNode}));
```

Comment:
Is it possible to have constants for these property names, in order to avoid errors/bugs due to human factor spelling mistake (dispatching event with name "current-ndoe" or similar)

Reply:
Will do.

I spent way too long trying to get typescript to do this so we could get typing of the event detail at the same time, but couldn't figure it out.